### DISCUSSION TITLE
how can I catch the end event of current task  in the plugin?

### DISCUSSION THREAD

**Discussion by zhengpq** [gulpjs/gulp#2609] [Help]
### What were you expecting to happen?

As the title say, i want to know how to listen the end event of current task in the plugin. I am developing a plugin that needs to delete some dir when the current task end. But i can not find a way  to do this besides to add an event listener in the task like the blow code

```js
function pluginTest() {
  return src(path)
    .pipe(newPlugin())
    .pipe(dest('dist'))
    .on('end', () => {})
}
```
What i  want is the plugin can know the end time of the task, so i can do the thing in the plugin, so i can use the plugin like this:

```js
function pluginTest() {
  return src(path)
    .pipe(newPlugin())
    .pipe(dest('dist'))
}
```

**Comment by phated** [comment]
I converted this to a discussion, where it belongs.
